extensions/amp-list/0.1/amp-list.js

```diff
@@ -340,40 +345,85 @@ export class AmpList extends AMP.BaseElement {
     );
   }
 
+  /**
+   * Returns true if element's src points to amp-state.
+   *
+   * @return {boolean}
+   * @private
+   */
+  isAmpStateSrc_() {
```

Comment:
Optional: Consider merging this with `getAmpStateJson_()` which always seem to be used together.

Reply:
I'd like to leave it separate for this PR and potentially refactor it in the followup where I add Promise/timeout support. AFAICT, they can't just be combined since I'd rather not rely on `getAmpStateJson_()` for control flow.